File: src/com/l2kt/gameserver/network/clientpackets/RequestSiegeDefenderList.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.data.manager.CastleManager;
import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.model.entity.Castle;
import com.l2kt.gameserver.network.serverpackets.SiegeDefenderList;

public final class RequestSiegeDefenderList extends L2GameClientPacket
{
	private int _castleId;
	
	@Override
	protected void readImpl()
	{
		_castleId = readD();
	}
	
	@Override
	protected void runImpl()
	{
		final Player activeChar = getClient().getActiveChar();
		if (activeChar == null)
			return;
		
		final Castle castle = CastleManager.getInstance().getCastleById(_castleId);
		if (castle == null)
			return;
		
		sendPacket(new SiegeDefenderList(castle));
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestSiegeDefenderList.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.data.manager.CastleManager
import com.l2kt.gameserver.network.serverpackets.SiegeDefenderList

class RequestSiegeDefenderList : L2GameClientPacket() {
    private var _castleId: Int = 0

    override fun readImpl() {
        _castleId = readD()
    }

    override fun runImpl() {
        client.activeChar ?: return

        val castle = CastleManager.getInstance().getCastleById(_castleId) ?: return

        sendPacket(SiegeDefenderList(castle))
    }
}